2022/day01/luigibarbato/parts.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"sort"
+	"strconv"
+)
+
+func PartOne(input []string) (int, error) {
+	buff := 0
+	arr := make([]int, 0)
+
+	for _, v := range input {
+		if v == "" {
+			arr = append(arr, buff)
+			buff = 0
+
+			continue
+		}
+
+		number, err := strconv.Atoi(v)
+		if err != nil {
+			return 0, err
+		}
+
+		buff += number
+	}
+
+	sort.Ints(arr)
```

Comment:
Instead of getting the last element you can use the `sort.Sort(sort.Reverse(sort.IntSlice(arr)))`, or use the sort.Slice

```go
sort.Slice(arr, func(a, b int) bool {
    return arr[b] < arr[a]
})
```

so that is clearer what you're looking for.

Reply:
For how badly Go handles the sort DESC, I'd argue that it's far clearer sorting ASC and take the last element :smiley: